x-pack/plugins/cases/public/components/all_cases/assignees_filter.tsx

```diff
@@ -0,0 +1,127 @@
+import { useAvailableCasesOwners } from '../app/use_available_owners';
+import { useCasesContext } from '../cases_context/use_cases_context';
+import { CurrentUserProfile } from '../types';
+import { EmptyMessage } from '../user_profiles/empty_message';
+import { NoMatches } from '../user_profiles/no_matches';
+import { SelectedStatusMessage } from '../user_profiles/selected_status_message';
+import { bringCurrentUserToFrontAndSort } from '../user_profiles/sort';
+import * as i18n from './translations';
+
+export interface AssigneesFilterPopoverProps {
+  selectedAssignees: UserProfileWithAvatar[];
+  currentUserProfile: CurrentUserProfile;
+  isLoading: boolean;
+  onSelectionChange: (users: UserProfileWithAvatar[]) => void;
+}
+
+const AssigneesFilterPopoverComponent: React.FC<AssigneesFilterPopoverProps> = ({
+  selectedAssignees,
+  currentUserProfile,
+  isLoading,
+  onSelectionChange,
+}) => {
+  const { owner: owners } = useCasesContext();
+  const hasOwners = owners.length > 0;
+  const availableOwners = useAvailableCasesOwners(['read']);
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const togglePopover = useCallback(() => setIsPopoverOpen((value) => !value), []);
+
+  const onChange = useCallback(
+    (users: UserProfileWithAvatar[]) => {
+      const sortedUsers = bringCurrentUserToFrontAndSort(currentUserProfile, users);
+      onSelectionChange(sortedUsers ?? []);
+    },
+    [currentUserProfile, onSelectionChange]
+  );
+
+  const selectedStatusMessage = useCallback(
+    (selectedCount: number) => (
+      <SelectedStatusMessage
+        selectedCount={selectedCount}
+        message={i18n.TOTAL_ASSIGNEES_FILTERED(selectedCount)}
+      />
+    ),
+    []
+  );
+
+  const onSearchChange = useCallback((term: string) => {
+    setSearchTerm(term);
+
+    if (!isEmpty(term)) {
+      setIsUserTyping(true);
+    }
+  }, []);
+
+  const [isUserTyping, setIsUserTyping] = useState(false);
+
+  const onDebounce = useCallback(() => setIsUserTyping(false), []);
+
+  const { data: userProfiles, isLoading: isLoadingSuggest } = useSuggestUserProfiles({
+    name: searchTerm,
+    owners: hasOwners ? owners : availableOwners,
+    onDebounce,
+  });
+
+  const searchResultProfiles = useMemo(
+    () => bringCurrentUserToFrontAndSort(currentUserProfile, userProfiles),
+    [userProfiles, currentUserProfile]
+  );
+
+  const isLoadingData = isLoading || isLoadingSuggest;
+
+  return (
+    <UserProfilesPopover
+      isOpen={isPopoverOpen}
+      closePopover={togglePopover}
+      panelStyle={{
+        minWidth: 520,
+      }}
+      button={
+        <EuiFilterButton
+          data-test-subj="options-filter-popover-button-assignees"
+          iconType="arrowDown"
+          onClick={togglePopover}
+          isLoading={isLoadingData}
+          isSelected={isPopoverOpen}
+          numFilters={userProfiles?.length ?? 0}
```

Comment:
The count shown when there are no active filters is a bit misleading I think - it currently seems to show the number of users that will be displayed in the popover on opening. Do you have access to the total number of user profiles? If not, can that badge be hidden if there are no active filters?

<img width="138" alt="image" src="https://user-images.githubusercontent.com/7405507/188854605-0bb23e46-adb9-4286-81a7-c15118affbcc.png">

Reply:
We chatted about this offline and we're going to hide this because we can't get the total number of profiles easily right now.